### DISCUSSION TITLE
How to use the No Results Handler

### DISCUSSION THREAD

**Discussion by anoonoll** [algolia/instantsearch#5297] [Q&A] [answered]
I would like to display a message when the number of searches is zero.
I have just read the page at the following link.
However, I could not figure out how to use the No Results Handler.
When I tried to import it from react-instantsearch-dom, I got an error.
How can I use the No Results Handler?

I am using React Instant Search v6.

https://www.algolia.com/doc/guides/building-search-ui/ecommerce-ui-template/components/in-depth/product-listing-page-display/no-results-handler/react/

**Comment by dhayab** [comment, marked as answer]
Hi, could you recreate your issue in a sandbox? Please note that this specific guide only works as-is in the [Ecommerce UI Template](https://www.algolia.com/doc/guides/building-search-ui/ecommerce-ui-template/getting-started/react/). To make it work in another React application you would have to adapt it considerably.

If you need to display a specific message when your query returns no results, you can also look at the [`<StateResults>`](https://www.algolia.com/doc/api-reference/widgets/state-results/react/) widget.

**Reply by anoonoll** [reply]
I just read the doc at StateResults, and I've successfully displayed the text in my React application when there are zero cases!
Thank you very much.

**Reply by dhayab** [reply]
Happy to help :)
